bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,257 @@
+        super.bridgeStatusChanged(bridgeStatusInfo);
+        if (thing.getStatus() == ThingStatus.ONLINE) {
+            initialize();
+        }
+    }
+
+    private boolean discoverAttributes(String localStation) {
+        SncfBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            Map<String, String> properties = new HashMap<>();
+            try {
+                bridgeHandler.stopPointDetail(localStation).ifPresent(stopPoint -> {
+                    String stationLoc = String.format("%s,%s", stopPoint.coord.lat, stopPoint.coord.lon);
+                    properties.put(LOCATION, stationLoc);
+                    properties.put(TIMEZONE, stopPoint.stopArea.timezone);
+                    PointType serverLoc = locationProvider.getLocation();
+                    if (serverLoc != null) {
+                        PointType stationLocation = new PointType(stationLoc);
+                        double distance = serverLoc.distanceFrom(stationLocation).doubleValue();
+                        properties.put(DISTANCE, new QuantityType<>(distance, SIUnits.METRE).toString());
+                    }
+                });
+                ThingBuilder thingBuilder = editThing();
+                thingBuilder.withProperties(properties);
+                updateThing(thingBuilder.build());
+                return true;
+            } catch (SncfException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+        return false;
+    }
+
+    private void scheduleRefresh(@Nullable ZonedDateTime when) {
+        // Ensure we'll try to refresh in one minute if no valid timestamp is provided
+        long wishedDelay = ZonedDateTime.now().until(when != null ? when : ZonedDateTime.now().plusMinutes(1),
+                ChronoUnit.SECONDS);
+        wishedDelay = wishedDelay < 0 ? 60 : wishedDelay;
+        logger.debug("wishedDelay is {} seconds", wishedDelay);
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            long existingDelay = job.getDelay(TimeUnit.SECONDS);
+            logger.debug("existingDelay is {} seconds", existingDelay);
+            if (existingDelay < wishedDelay && existingDelay > 0) {
+                logger.debug("Do nothing, existingDelay earlier than wishedDelay");
+                return;
+            }
+            freeRefreshJob();
+        }
+        logger.debug("Scheduling update in {} seconds.", wishedDelay);
+        refreshJob = scheduler.schedule(() -> updateThing(), wishedDelay, TimeUnit.SECONDS);
+    }
+
+    private void updateThing() {
+        SncfBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            scheduler.submit(() -> updatePassage(bridgeHandler, GROUP_ARRIVAL));
+            scheduler.submit(() -> updatePassage(bridgeHandler, GROUP_DEPARTURE));
```

Comment:
What annoys me with these calls called in parallel in separate threads is that you have to consider concurrent calls to `scheduleRefresh` and `freeRefreshJob`. I am not sure this is something you took into consideration, did you ?
Is it really necessary to do that ? Two calls in sequence could be ok, no ?

Reply:
Well, I had to do this because I faced a concurrent exception at jetty level with the previous successive calls. It is the only I found to solve the problem.

Reply:
I still do not understand why you create a new thread while `updateThing` is already executed by a scheduled thread ?

Reply:
If/when I don't do it in a thread, I get this error on second call : 
``
17:32:17.180 [DEBUG] [f.internal.handler.SncfBridgeHandler] - Execution interrupted : null
java.lang.InterruptedException: null
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1343) ~[?:?]
``